### PR TITLE
Do not allow importing by test target name in integration tests

### PR DESCRIPTION
**Pull request by maciektr**

### CODE REVIEW

File: scarb/src/compiler/db.rs [hunk 1]
```diff
@@ -169,6 +169,7 @@ fn build_project_config(unit: &CairoCompilationUnit) -> Result<ProjectConfig> {
                         .any(|target| {
                             target.group_id.clone().unwrap_or(target.name.clone())
                                 == component.package.id.name.to_smol_str()
+                            && component_as_dependency.cairo_package_name() != component.cairo_package_name()
```
Comment (DelevoXDG) [inline scarb/src/compiler/db.rs:172]: Why this change?

Comment (maciektr) [reply scarb/src/compiler/db.rs:172]: To make sure, we do not add dependency on self here.

File: scarb/src/compiler/db.rs [hunk 2]
```diff
@@ -184,13 +185,21 @@ fn build_project_config(unit: &CairoCompilationUnit) -> Result<ProjectConfig> {
...
 
             // Adds itself to dependencies
-            dependencies.insert(
-                component.package.id.name.to_string(),
-                DependencySettings {
-                    version: (component.package.id.name.to_string() != *CORELIB_CRATE_NAME)
-                        .then_some(component.package.id.version.clone()),
-                },
-            );
+            if component.first_target().kind.is_test() {
```
Comment (DelevoXDG) [inline scarb/src/compiler/db.rs:188]: Correct me if I'm wrong, but the title of this PR and previous state of things suggest we only want to filter out integration tests, not all non-tests?

Comment (maciektr) [reply scarb/src/compiler/db.rs]: Good catch! Submitted a fix :D

**Review by DelevoXDG — APPROVED** [review]

**Comment by piotmag769** [comment]
@enitrat a heads up cause I know you use this import method in integration tests for some projects: pls migrate to `use crate::`

**Comment by enitrat** [comment]
thanks 🙏
